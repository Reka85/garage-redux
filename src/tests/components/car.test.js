import React from "react";
import  { shallow, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
configure({ adapter: new Adapter() });
import Car from "../../components/car";

test("should render Car correctly", () => {
  const wrapper = shallow(<Car brand={"Ford"} model={"456"} owner={"Joe"} plate={"ABC123"}/>);
  expect(wrapper).toMatchSnapshot();
});
