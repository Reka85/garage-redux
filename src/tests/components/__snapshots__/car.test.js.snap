// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Car correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Car
    brand="Ford"
    model="456"
    owner="Joe"
    plate="ABC123"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <div
        className="car-item"
      >
        <h4>
          Ford
           - 
          456
        </h4>
        <p>
          Owner: 
          Joe
        </p>
        <p>
          Plate number: 
          <span
            className="plate"
          >
            ABC123
          </span>
        </p>
      </div>,
      "replace": false,
      "to": "cars/undefined",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h4>
            Ford
             - 
            456
          </h4>,
          <p>
            Owner: 
            Joe
          </p>,
          <p>
            Plate number: 
            <span
              className="plate"
            >
              ABC123
            </span>
          </p>,
        ],
        "className": "car-item",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Ford",
              " - ",
              "456",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Ford",
            " - ",
            "456",
          ],
          "type": "h4",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Owner: ",
              "Joe",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Owner: ",
            "Joe",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Plate number: ",
              <span
                className="plate"
              >
                ABC123
              </span>,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Plate number: ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "ABC123",
                "className": "plate",
              },
              "ref": null,
              "rendered": "ABC123",
              "type": "span",
            },
          ],
          "type": "p",
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <div
          className="car-item"
        >
          <h4>
            Ford
             - 
            456
          </h4>
          <p>
            Owner: 
            Joe
          </p>
          <p>
            Plate number: 
            <span
              className="plate"
            >
              ABC123
            </span>
          </p>
        </div>,
        "replace": false,
        "to": "cars/undefined",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h4>
              Ford
               - 
              456
            </h4>,
            <p>
              Owner: 
              Joe
            </p>,
            <p>
              Plate number: 
              <span
                className="plate"
              >
                ABC123
              </span>
            </p>,
          ],
          "className": "car-item",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Ford",
                " - ",
                "456",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Ford",
              " - ",
              "456",
            ],
            "type": "h4",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Owner: ",
                "Joe",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Owner: ",
              "Joe",
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Plate number: ",
                <span
                  className="plate"
                >
                  ABC123
                </span>,
              ],
            },
            "ref": null,
            "rendered": Array [
              "Plate number: ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "ABC123",
                  "className": "plate",
                },
                "ref": null,
                "rendered": "ABC123",
                "type": "span",
              },
            ],
            "type": "p",
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
